web/annotation_layer_builder.js

```diff
@@ -195,6 +209,23 @@ class AnnotationLayerBuilder {
     return !!this.annotationLayer?.hasEditableAnnotations();
   }
 
+  async injectLinkAnnotations(inferredLinks, viewport, structTreeLayer) {
```

Comment:
- This should take a parameter object (and it needs a JSDoc-typedef as well), similar to how various `render`-methods were recently updated.

 - Also, this must *obviously* check that the annotationLayer is still active.
```suggestion
  async injectLinkAnnotations({ inferredLinks, viewport, structTreeLayer = null }) {
    if (this._cancelled) {
      return;
    }
```